### PR TITLE
WT-13525 Migrating to AWS-based MacOS hosts

### PR DESCRIPTION
**Pull request by bzm-jas**
DevProd are migrating our MacOS hosts from MscStadium to AWS so we are making changes in WT evergreen configuration to switch from macos-13-arm64 to macos-14-arm64. Our compilation requires the swig package, which is no longer installed in macos by default. Therefore I have modified the script for configure_wiredtiger and make_wiredtiger to install swig as a temporary workaround. A ticket has been filed to install siwg on our macos-14-arm64 distro and the temporary workaround will be removed after that ticket is completed.

### CODE REVIEW

File: test/evergreen.yml
Comment (jiechenbo) [inline test/evergreen.yml:273]: np: I would remove DEVPROD-12418 from the comment as it is internal to MongODB. The ticket WT-13704 should be enough and is public facing.

File: test/evergreen.yml
Comment (jiechenbo) [inline test/evergreen.yml:265]: It seems we only use macos-14-arm64, it might be worth considering to change the existing code to that. I wonder what happened.

Comment (jiechenbo) [reply test/evergreen.yml:265]: Update: https://jira.mongodb.org/browse/WT-12595 I have found the reason why. Disregard this comment.

File: test/evergreen.yml
Comment (bzm-jas) [reply test/evergreen.yml:273]: sounds good, I've removed the devprod ticket number in: [6017ba5](https://github.com/wiredtiger/wiredtiger/pull/11174/commits/6017ba5be49b1189005fd3ed67153774adcb2932)

**Review by jiechenbo — APPROVED** [review]
LGTM! Please fill in the section by the bot and perform another patch build at the end before merging

**Comment by jeremythorp** [comment]
Hi @bzm-jas , as Mac tests are not part of the default PR testing, please can you manually launch the Mac tests within the PR testing. Thanks, Jeremy

**Review by jeremythorp — APPROVED** [review]
The Mac PR tests have passed, so LGTM!

**Pull request merged**
